Mathlib/LinearAlgebra/Eigenspace/Matrix.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.LinearAlgebra.Eigenspace.Basic
+
+/-!
+
+# Eigenvalues, Eigenvectors and Spectrum for Matrices
+
+This file collects results about eigenvectors, eigenvalues and spectrum specific to matrices
+over a field.
+
+## Tags
+eigenspace, eigenvector, eigenvalue, spectrum, matrix
+
+-/
+
+section SpectrumDiagonal
+
+variable {R n M : Type*} [DecidableEq n][Fintype n]
+
+open Matrix
+open Module.End
+
+section NontrivialCommRing
+
+variable [CommRing R] [Nontrivial R] [AddCommGroup M] [Module R M]
+
+/-- Basis vectors are eigenvectors of associated diagonal linear operator. -/
+lemma hasEigenvector_toLin_diagonal (d : n → R) (i : n) (b : Basis n R M) :
+    HasEigenvector (toLin b b (diagonal d)) (d i) (b i) :=
+  ⟨mem_eigenspace_iff.mpr <| by simp [diagonal], Basis.ne_zero b i⟩
+
+/--  Standard basis vectors are eigenvectors of any associated diagonal linear operator. -/
+lemma hasEigenvector_toLin'_diagonal (d : n → R) (i : n) :
+    HasEigenvector (toLin' (diagonal d)) (d i) (Pi.basisFun R n i)  :=
+  hasEigenvector_toLin_diagonal ..
+
+section NoZeroSMulDivisors
+
+variable [NoZeroSMulDivisors R M]
```

Comment:
I wouldn't create a section for one result, just add this hypothesis to the lemma